zerver/lib/zulip_update_announcements.py

```diff
@@ -456,6 +456,8 @@ def send_zulip_update_announcements_to_realm(
             timezone_now() - level_none_to_initial_auditlog.event_time < timedelta(days=7)
         ):
             new_zulip_update_announcements_level = latest_zulip_update_announcements_level
+    elif realm.zulip_update_announcements_stream.deactivated:
+        new_zulip_update_announcements_level = latest_zulip_update_announcements_level
```

Comment:
Should this be `elif realm.zulip_update_announcements_stream is None or realm.zulip_update_announcements_stream.deactivated`? 

I'm not sure whether the `level_none_to_initial_auditlog` check should be applied here.

Reply:
I initially did that but decided to use the current approach because if the configured stream is deactivated then `level_none_to_initial_auditlog` check doesn't play any meaningful role:
* `if level_none_to_initial_auditlog is None`: This is for a new realm created. If the configured stream was later manually deactivated then we can simply skip update messages. No 1 week wait time kind of thing makes sense.
* `if timezone_now() - level_none_to_initial_auditlog.event_time < timedelta(days=7)`: This is for Old realms or realm imported from other product. For old realms (predating this feature), 1 week since group DM was sent has already passed. For realms imported from other products, they need to set the stream -- and if they later deactivate that we can simply skip messages.

Reply:
Hmm, there's one case where it'd be useful. If a realm (predating this feature) which was deactivated now gets reactivated. Then if the configured stream is deactivated we'd like to wait for a week after sending group DM.

But I wonder if such a realm exists as we must have a policy to remove deactivated realms from our db periodically ? (even if `scheduled_deletion_date` is not configured during deactivation)

Reply:
Weird corner cases like that are certainly possible, especially with self-hosted servers that may upgrade very late. Nothing forces deactivated realms to ever be deleted in the general self-hosted context.

Reply:
https://chat.zulip.org/#narrow/channel/3-backend/topic/deactivated.20zulip_update_announcements_stream/near/2026827